components/script/dom/globalscope.rs

```diff
@@ -1251,18 +1330,59 @@ impl GlobalScope {
     }
 
     fn read_file(&self, id: Uuid) -> Result<Vec<u8>, ()> {
+        let recv = self.send_msg(id);
+        GlobalScope::read_msg(recv)
+    }
+
+    pub fn read_file_async(
+        &self,
+        id: Uuid,
+        promise: Rc<Promise>,
+        callback: Box<dyn Fn(Rc<Promise>, Result<Vec<u8>, Error>) + Send>,
```

Comment:
nit: here we can use `FileListenerCallback`

Reply:
Actually, please discard this comment, we already type the callback below when we add it to the file listener, and using `FileListenerCallback` in the function signature directly will just make its use by the blob more complicated. 

So just the space with the doc comment, and it's good to go!